RecoMuon/TrackerSeedGenerator/plugins/MuonHLTSeedMVAClassifier.cc

```diff
@@ -0,0 +1,286 @@
+
+// Package:    RecoMuon_TrackerSeedGenerator
+// Class:      MuonHLTSeedMVAClassifier
+
+// Original Author:  Won Jun, OH Minseok
+//         Created:  Fri, 28 May 2021
+
+// system include files
+#include <memory>
+#include <cmath>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+// Geometry
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+// TrajectorySeed
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeed.h"
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+#include "DataFormats/TrajectorySeed/interface/PropagationDirection.h"
+#include "DataFormats/TrajectoryState/interface/PTrajectoryStateOnDet.h"
+#include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+#include "DataFormats/TrackingRecHit/interface/TrackingRecHit.h"
+
+#include "RecoMuon/TrackerSeedGenerator/interface/SeedMvaEstimator.h"
+
+// class declaration
+bool sortByMvaScore(const std::pair<unsigned, double>& A, const std::pair<unsigned, double>& B) {
+  return (A.second > B.second);
+};
+
+class MuonHLTSeedMVAClassifier : public edm::stream::EDProducer<> {
+public:
+  explicit MuonHLTSeedMVAClassifier(const edm::ParameterSet&);
+  ~MuonHLTSeedMVAClassifier() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  // member data
+  edm::EDGetTokenT<TrajectorySeedCollection> t_Seed_;
+  edm::EDGetTokenT<l1t::MuonBxCollection> t_L1Muon_;
+  edm::EDGetTokenT<reco::RecoChargedCandidateCollection> t_L2Muon_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> t_trackerGeometry_;
+
+  typedef std::pair<std::unique_ptr<const SeedMvaEstimator>, std::unique_ptr<const SeedMvaEstimator>>
+      pairSeedMvaEstimator;
+  pairSeedMvaEstimator mvaEstimator_;
+
+  edm::FileInPath mvaFileB_;
+  edm::FileInPath mvaFileE_;
+
+  std::vector<double> mvaScaleMeanB_;
+  std::vector<double> mvaScaleStdB_;
+  std::vector<double> mvaScaleMeanE_;
+  std::vector<double> mvaScaleStdE_;
+
+  const double etaEdge_;
+  const double mvaCutB_;
+  const double mvaCutE_;
+
+  const bool doSort_;
+  const int nSeedsMaxB_;
+  const int nSeedsMaxE_;
+
+  const bool rejectAll_;
+  const bool isFromL1_;
+  const int minL1Qual_;
+  const double baseScore_;
+
+  double getSeedMva(const pairSeedMvaEstimator& pairMvaEstimator,
```

Comment:
`minL1Qual_` , `isFromL1_` , `baseScore_` are all (const) class members: there is no need to pass them as arguments to a private method.